core/src/main/java/com/sequenceiq/cloudbreak/service/AbstractWorkspaceAwareResourceService.java

```diff
@@ -76,6 +79,22 @@ public T getByNameForWorkspace(String name, Workspace workspace) {
         return object;
     }
 
+    @Override
+    public Set<T> getByNamesForWorkspaceId(Set<String> names, Long workspaceId) {
```

Comment:
First of all please import idea inspections or take care on them, because this new methods contains warnings.
That foreach is a `closure` and uses variables outside of the lambda. Would be better to avoid closure if possible.

There are two ways, use `notFound.stream().collect(Collectors.joining(", "))`, but more simply is `String.join(", ", notFound)`
```
throw new NotFoundException(String.format("No %s(s) found with name(s) '%s'", resource().getShortName(), String.join(", ", notFound)));
```

Reply:
I can do this, but is there any way to automate these inspections with checkstyle? @bhavanki doesn't use IntelliJ, so this would be impossible for him.

I elected to do the joining style I did to maintain parity with existing NotFoundExceptions wrapping the names in single quotes. I can eliminate this, but as long as it's okay to break that similarity in behavior. Additionally, I can use String.join instead of the StringJoiner.

Is there a style guide?

Reply:
The team is using Intellij, so @bhavanki's responsible to match code style with our standards. Unfortunately it isn't possible to integrate Intellij idea inspections with our pipeline. Our pipeline contains the basics, and intellij gives tips to make the code better.

We don't have "style guide", but you can check which inspections are turned on. We spent lot time to figure out the best fit with our project.

Reply:
We do have a style guide that is drafted and out for discussion. We can include whether we should avoid closures. I personally see no reason to avoid them, and wonder why we are expending the effort to move to the latest JDK if we don't want to use newer language features.

I don't use IntelliJ. I can try to add in its custom annotations or whatever based on patterns that I see, but I'm not going to alter my work style merely to access their walled garden. If they are important, then they should be covered in the build pipeline like checkstyle and spotbugs in a way that anyone can benefit from them.

Reply:
"If they are important" they are important, mostly because we can make the code bug less and more readable by following them. But those are not black and white like checkstyle issues. There are inspections which has meaning in some context and not in an other. So the developers have check them code warnings. I suggest you to run code analysis once with idea (exclude tests to get better picture) and you will see the result (we don't have to fix all of them, but many of them are valid inspections). I don't say that you have to use Intellij, if you can write the most simple and well behavior code without IDE help, go ahead. But if you check the concrete situation, the code is valid and do the right thing, but there is a better way to solve it, that is where inspections help us.

Reply:
" if we don't want to use newer language features" i haven't understand this thing, could you please explain?

Reply:
I think I got this fix in. At least IDEA inspections didn't bring anything up here. Let me know what you think.